library/lua/gui/widgets.lua

```diff
@@ -235,9 +235,13 @@ end
 function render_text(obj,dc,x0,y0,pen,dpen,disabled)
     local width = 0
     for iline,line in ipairs(obj.text_lines) do
```

Comment:
I think changing this to a numeric for loop might lead to slightly cleaner logic than the current approach (continuing until the desired start index is reached), e.g.:
```lua
for iline in (obj.start_line_num or 1), #obj.text_lines do
  local line = obj.text_lines[iline]
```
(I'm not quite sure what the `dc` check is for, but it might also need to be included in the first expression there)

Reply:
done. the dc check is because render_text is called with nil dc from Label:getTextWidth() and List:getContentWidth(), and in those cases it is important to let it "render" all the text so obj.text_width can be set at the end of the function

Reply:
Ah, thanks for the explanation